### PR TITLE
Domain test charge integration events

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Add tests of published charge integration events.
Refactors - including splitting - existing domain tests.

This PR will be succeeded by new PRs covering:
- Use correlation ID to track whether e.g. published events correlate with the triggering command/event
- Optimize code in order to make test code shorter and more maintainable

## References

* #716

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/Charges/ChargeIngestionTests.cs [hunk 1]
```diff
@@ -78,19 +86,90 @@ public async Task When_CallingChargeIngestion_Then_RequestIsProcessedAndMessageI
...
 
                 // Act
-                var actualResponse = await Fixture.HostManager.HttpClient.SendAsync(request);
+                await Fixture.HostManager.HttpClient.SendAsync(request);
 
                 // Assert
-                using var assertionScope = new AssertionScope();
+                var isMessageReceived = isMessageReceivedEvent.Wait(TimeSpan.FromSeconds(10));
+                isMessageReceived.Should().BeTrue();
+                body.Should().NotBeEmpty();
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/Charges/ChargeIngestionTests.cs:94]: No check for correlationId?

Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/Charges/ChargeIngestionTests.cs:94]: Next PR according to description (perhaps I added it after you started reviewing 😄)

**Review by HenrikSommer — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeLinks/MessageHub/ChargeLinkDataAvailableNotifierEndpointTests.cs [hunk 2]
```diff
@@ -92,13 +92,15 @@ public async Task When_ReceivingChargeLinkMessage_MessageHubIsNotifiedAboutAvail
 
                 // => Was data available notification sent from the domain?
                 //    (Timeout should not be more than 5 secs. - currently it's high so we can break during demo.)
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeLinks/MessageHub/ChargeLinkDataAvailableNotifierEndpointTests.cs:94]: when is this demo'ed? why not set it to 5 now?

**Review by lasrinnil — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeLinks/MessageHub/ChargeLinkDataAvailableNotifierEndpointTests.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeLinks/MessageHub/ChargeLinkDataAvailableNotifierEndpointTests.cs:94]: It's my intention to fix this for all domain tests in one of the succeeding PRs.

**Pull request merged**